src/mbgl/gl/program.hpp

```diff
@@ -37,6 +41,57 @@ class Program {
           attributeLocations(Attributes::loadNamedLocations(binaryProgram)),
           uniformsState(Uniforms::loadNamedLocations(binaryProgram)) {
     }
+    
+    template <class Shaders>
```

Comment:
To reduce binary size, pass `vertexSource`, `fragmentSource`, and `name` as parameters rather than templating on `Shaders`.